### PR TITLE
integration-tests: properly stop snapd from branch

### PR DESCRIPTION
**Pull request by fgimenez**
Requires #538 

With these changes when building from branch we manage the snapd process using the systemd tools, being able to start and stop the service with the built binary properly, and restoring the system's service once the test is over.

### CODE REVIEW

File: integration-tests/tests/base_test.go [hunk 1]
```diff
@@ -64,10 +67,16 @@ func Test(t *testing.T) {
 		os.Stdout,
 		report.NewSubunitV2ParserReporter(&report.FileReporter{}))
 	runner.TestingT(t, output)
+
+	if daemonParentCmd != nil && daemonParentCmd.Process != nil {
+		if err := stopDaemonProcess("snapd"); err != nil {
+			fmt.Printf("Error stopping daemon: %v\n", err)
```
Comment (niemeyer) [inline integration-tests/tests/base_test.go]: s/fmt.Printf/t.Fatalf/

**Comment by niemeyer** [comment]
LGTM, with one trivial note.

**Comment by mvo5** [comment]
Thanks for this branch. The code looks just fine. But I wonder if we might get away with something slightly simpler? It seems like `systemctl kill -s SIGINT ubuntu-snappy.snapd.service` should also work and that means we can avoid the extra getPidFromName() parsing? Or am I missing something?

**Comment by fgimenez** [comment]
@mvo5 it seems that, being created by systemd-activate, the snapd process cannot be properly stopped referring to `ubuntu-snappy.snapd.service`, the only method that has worked for me so far is getting the pid and sending the signal to it directly, don't know if there might be another options.

Thanks! :)

**Comment by mvo5** [comment]
Oh, silly me. Of course the key is systemd-activate here. 

The code is fine, I'm still a little bit unhappy about the pid extraction, it does not feel good.

I see two alternatives:

(1) Running it inside a named systemd scope (cgroup):
```
root@localhost: ~# systemd-run --scope --unit=foo /lib/systemd/systemd-activate -l /run/snapd.socket ./snapd  &
Running scope as unit foo.scope.
Listening on /run/snapd.socket as 3.

root@localhost:~# systemctl status foo.scope|grep Active
   Active: active (running) since Mon 2016-02-29 10:12:27 UTC; 8s ago
root@localhost:~# systemctl stop foo.scope
Child 1512 died with code 1
[1]+  Terminated              systemd-run --scope --unit=foo /lib/systemd/systemd-activate -l /run/snapd.socket ./snapd
root@localhost:~# systemctl status foo.scope|grep Active
   Active: inactive (dead)
```

(2) Bind mounting our copy over the real thing and use the standard commands:
```
# systemctl stop ubuntu-snappy.snapd.service
# mount -o bind new-snapd /usr/bin/snapd
# systemctl start ubuntu-snappy.snapd.service
```

But I don't want to block this branch, it brings us one step closer to our goal :) I just hope we can find something more elegant than poking into the systemd journal to find the right pid (and given that the systemd interfaces were not always that stable there is a certain risk of breakage with new systemd versions).

File: integration-tests/tests/base_test.go (hunk 1 above)
Comment (fgimenez) [inline integration-tests/tests/base_test.go]: @niemeyer very good catch, integrated into the new approach

**Comment by fgimenez** [comment]
@mvo5 both alternatives look much cleaner than the previous approach :) I'm having problems with the first one, maybe I'm doing something wrong but it seems that the daemon is not listening after calling systemd-run, can you verify if it works in your example?

I've pushed changes that implement the second one, but it's still not working, it hangs when trying to stop the service. Also, could you check if the `SNAPPY_TRUSTED_ACCOUNT_KEY` is properly set when starting it?

Thanks!

**Comment by fgimenez** [comment]
@mvo5 sorry, after rebasing to master the problem stopping the service is gone :)

I've checked that snapAckSuite, which use `SNAPPY_TRUSTED_ACCOUNT_KEY`, fails with the current code, something is still wrong with how that environment variable is set.

Also, do you think we should umount the built binary and restart the original service after finishing the test?

Thanks!

**Comment by fgimenez** [comment]
@mvo5 I've pushed the changes for writing the environment info to `/etc/systemd/system/ubuntu-snappy.snapd.service.d/env.conf` and issuing `systemctl daemon-reload` calls before starting the daemon, all seems to be working now, please take another look :)

Thanks!

**Comment by mvo5** [comment]
@fgimenez Great work! Thanks a bunch. And I do feel slightly bad because in the end my suggestion was more work for you (and some of it was a bit tricky indeed). But I like the result :) :+1:

**Comment by chipaca** [comment]
:+1:

**Pull request merged**